src/main/scala/nl.knaw.dans.easy.deposit/docs/DatasetMetadata.scala

```diff
@@ -61,83 +58,31 @@ case class DatasetMetadata(identifiers: Option[Seq[SchemedValue[String]]] = None
     case SchemedValue(`doiScheme`, value) => value
   })
 
-  def setDoi(value: String): DatasetMetadata = {
-    val ids = identifiers.getOrElse(Seq.empty).filter(_.scheme == doiScheme)
-    this.copy(identifiers = Some(ids :+ SchemedValue(doiScheme, value)))
-  }
-
-  private lazy val submitDate: Option[String] = {
-    dates.flatMap(_.find(_.qualifier == DateQualifier.dateSubmitted)).map(_.value)
+  lazy val privacyBoolean: Try[Boolean] = privacySensitiveDataPresent match {
+    case PrivacySensitiveDataPresent.yes => Success(true)
+    case PrivacySensitiveDataPresent.no => Success(false)
+    case PrivacySensitiveDataPresent.unspecified => Failure(missingValue("PrivacySensitiveDataPresent"))
   }
 
-  lazy val xml: Try[Elem] = Success(<stub/>) // TODO
+  lazy val licenceAccepted: Try[Unit] = if (acceptLicenseAgreement) Success(())
```

Comment:
great job! I like this style! Would it make sense to actually let this be the entry point of `acceptLicenseAgreement` and make that field private? Now you have two entry points to that field...

Reply:
Would json4s deserialize private fields?

Reply:
note sure, but feel free to test it 😉